File: JavaSource/src/main/java/ru/akhcheck/patterns/mediator/vehicles/Vehicle.java
package ru.akhcheck.patterns.mediator.vehicles;

import ru.akhcheck.patterns.mediator.dispatcher.Dispatcher;

/**
 * Участник, который общается только через посредника.
 *
 * <p>Ключевое здесь то, чего нет: у самолёта нет ссылок на другие самолёты. Он
 * знает лишь диспетчера, поэтому добавление десятого самолёта не меняет ни
 * одного из девяти существующих.
 */
public abstract class Vehicle {
    private final String name;
    private Dispatcher dispatcher;

    protected Vehicle(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public void setDispatcher(Dispatcher dispatcher) {
        this.dispatcher = dispatcher;
    }

    protected boolean notifyDispatcher(String message) {
        if (dispatcher == null) {
            throw new IllegalStateException(name + " has no dispatcher");
        }
        return dispatcher.notify(this, message);
    }
}

File: Mediator/python-source/vehicles/Vehicle.py
from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from dispatcher import Dispatcher


class Vehicle:
    """Участник, который общается только через посредника.

    Ключевое здесь то, чего нет: у самолёта нет ссылок на другие самолёты. Он
    знает лишь диспетчера, и потому добавление десятого самолёта не меняет ни
    одного из девяти существующих.
    """

    def __init__(self, name: str) -> None:
        self.name = name
        self._dispatcher: Dispatcher | None = None

    def set_dispatcher(self, dispatcher: Dispatcher) -> None:
        self._dispatcher = dispatcher

    def notify(self, message: str) -> bool:
        if self._dispatcher is None:
            raise ValueError(f"{self.name} has no dispatcher")
        return self._dispatcher.notify(self, message)
